dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+    private static final String ORACLE_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID    = "alter table workflow_action_step add constraint fk_workflow_action_step_step_id   foreign key (step_id)   references workflow_step  (id)";
+
+    private static final String H2_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEY_ACTION_ID       = "alter table workflow_action_step add constraint fk_workflow_action_step_action_id foreign key (action_id) references workflow_action(id)";
+    private static final String H2_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID        = "alter table workflow_action_step add constraint fk_workflow_action_step_step_id   foreign key (step_id)   references workflow_step  (id)";
+
+    @Override
+    public boolean forceRun() {
+        return Boolean.TRUE;
+    }
+
+    @Override
+    public void executeUpgrade() throws DotDataException {
+        final DotConnect dc = new DotConnect();
+        if (DbConnectionFactory.isMsSql()) {
+            try {
+                dc.executeStatement("SET TRANSACTION ISOLATION LEVEL READ COMMITTED");
+            } catch (SQLException e) {
+                throw new DotRuntimeException(
+                        "Transaction isolation level could not be set.", e);
+            }
+        }
+
+        // SCHEMA CHANGES
+        this.createWorkflowActionStepTable     (dc);
+        this.addSchemeIdColumn                 (dc);
+        //this.addRequiresCheckoutOptionColumn   (dc);
+
+        // DATA CHANGES
+        this.addWorkflowActionStepData         (dc);
+        this.updateWorkflowActionData          (dc);
+        this.updateWorkflowSchemeXStructureData(dc);
+    } // executeUpgrade.
+
+    private void updateWorkflowSchemeXStructureData(DotConnect dc) throws DotDataException {
+        Logger.info(this, "Updating index in 'workflow_scheme_x_structure' table.");
+        try {
+            dc.setSQL(selectTableIndex());
+            final List<Map<String, Object>> tableIndex = dc.loadObjectResults();
+            if (!tableIndex.isEmpty()) {
+                dc.executeStatement(dropTableIndex());
+            }
+            // In MySQL and SQL Server re-creating the index is NOT necessary as the regular index
+            // is created separately
+            if (DbConnectionFactory.isPostgres() || DbConnectionFactory.isOracle()) {
+                dc.executeStatement(createTableIndex());
+            }
+        } catch (SQLException e) {
+            throw new DotRuntimeException(
+                    "An error occurred when updating the index in 'workflow_scheme_x_structure' table.",
+                    e);
+        }
+    }
+
+    private void updateWorkflowActionData(final DotConnect dc) throws DotDataException {
+
+        Logger.info(this, "Associating Workflow Actions to Workflow Schemes.");
+
+        final List<Map<String, Object>> schemeIds =
+                dc.setSQL(selectSchemeIdsForActions()).loadObjectResults();
+        schemeIds.stream().forEach(row -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'row' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)